graphbolt/include/graphbolt/rowwise_pick.h

```diff
@@ -0,0 +1,184 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/rowwise_pick.h
+ * @brief Contains the function definition for RowWisePickPerEtype.
+ */
+
+#ifndef GRAPHBOLT_ROWWISE_PICK_H_
+#define GRAPHBOLT_ROWWISE_PICK_H_
+
+#include <graphbolt/csc_sampling_graph.h>
+#include <graphbolt/utils.h>
+
+using namespace graphbolt::utils;
+
+namespace graphbolt {
+namespace sampling {
+
+using CSCPtr = CSCSamplingGraph*;
+using RangePickFn = std::function<torch::Tensor(
```

Comment:
This is exactly the same RangePickFn as graphbolt/include/graphbolt/utils.h, we should not redefine it.

Reply:
How aout RangedPicker?

Reply:
My genuine worry is the return of a PyTorch Tensor will induce costly memory allocation, especially since the functor is intended to be called by large number of threads. If some parts of PyTorch's underlying memory management acquires locking, the speed will be hugely impacted. As a comparison, the original DGL implementation does the following:

1. It allocates a tensor to store the number of samples of each row.
2. It calculates the number of samples using a separate functor (in parallel).
3. Conduct prefix sum to get the size of the output buffer and the start and end offset of each row.
4. Allocate the output buffer at once.
5. Invokes the user-defined pick functor which directly writes to the output buffer.

Reply:
Changed to this workflow and exaplain more in docstring.